lib/hwm/hwm.go

```diff
@@ -120,6 +139,15 @@ func (trie *HighWatermarkTracker) DeleteOldKeys(hwmLimit uint64) []string {
 	return deletedKeys
 }
 
+func (trie *HighWatermarkTracker) ToMap() map[string]uint64 {
```

Comment:
It's sad that this function is only used by the tests.  Does it have to be here?